lib/storage/src/rbac/ops_checks.rs

```diff
@@ -457,3 +452,773 @@ impl PayloadConstraint {
+                .add("col", false, false)
+                .add("col2", false, true)
+                .into(),
+        );
+        assert_forbidden(
+            &DiscoverRequestInternal {
+                context: Some(vec![ContextExamplePair {
+                    positive: RecommendExample::Dense(vec![0.0, 1.0, 2.0]),
+                    negative: RecommendExample::PointId(ExtendedPointId::NumId(12345)),
+                }]),
+                ..op.clone()
+            },
+            &AccessCollectionBuilder::new()
+                .add("col", false, false)
+                .add("col2", false, true)
+                .into(),
+        );
+
+        // lookup_from requires read access
+        assert_forbidden(
+            &op,
+            &AccessCollectionBuilder::new()
+                .add("col", false, true)
+                .into(),
+        );
+        assert_allowed(
+            &DiscoverRequestInternal {
+                lookup_from: None,
+                ..op.clone()
+            },
+            &AccessCollectionBuilder::new()
+                .add("col", false, true)
+                .into(),
+        );
+    }
+
+    #[test]
+    fn check_test_scroll_request_internal() {
+        let op = ScrollRequestInternal {
+            offset: Some(ExtendedPointId::NumId(12345)),
+            limit: Some(100),
+            filter: None,
+            with_payload: Some(WithPayloadInterface::Bool(true)),
+            with_vector: WithVector::Bool(true),
+            order_by: Some(OrderByInterface::Key("path".parse().unwrap())),
+        };
+
+        assert_allowed(&op, &Access::Global(GlobalAccessMode::Manage));
+        assert_allowed(&op, &Access::Global(GlobalAccessMode::Read));
+
+        assert_allowed(
+            &op,
+            &AccessCollectionBuilder::new()
+                .add("col", false, true)
+                .into(),
+        );
+
+        assert_allowed_rewrite(
+            &ScrollRequestInternal { ..op.clone() },
+            &AccessCollectionBuilder::new()
+                .add("col", false, false)
+                .into(),
+            |op| {
+                op.filter = Some(PayloadConstraint::new_test("col").to_filter());
+            },
+        );
+    }
+
+    #[test]
+    fn test_collection_update_operations() {
+        CollectionUpdateOperationsDiscriminants::iter().for_each(|discr| match discr {
+            CollectionUpdateOperationsDiscriminants::PointOperation => {
+                test_collection_update_operations_points()
+            }
+            CollectionUpdateOperationsDiscriminants::VectorOperation => {
+                test_collection_update_operations_update_vectors()
+            }
+            CollectionUpdateOperationsDiscriminants::PayloadOperation => {
+                test_collection_update_operations_payload()
+            }
+            CollectionUpdateOperationsDiscriminants::FieldIndexOperation => {
+                test_collection_update_operations_field_index()
+            }
```

Comment:
I suggest to use `check_` prefix instead of `test_` for these sub_tests, since they aren't technically "standalone" tests